### PR TITLE
feat: add option for base URL

### PR DESCRIPTION
**Pull request by nicolindemann**
Since hosting.de and http.net share the same API (see https://www.http.net/docs/api/?json), this PR is meant to be able to use this provider for http.net as well. Unfortunately, I have only limited experience with Go - if any of the approaches are wrong/inappropriate I would be happy to receive feedback.

### CODE REVIEW

**Comment by teutat3s** [comment]
Thank you for this contribution. I don't have time to look at this right now, but I'll respond with a proper review soon.

File: hostingde/provider.go [hunk 1]
```diff
@@ -102,6 +108,15 @@ func (p *hostingdeProvider) Configure(ctx context.Context, req provider.Configur
 		auth_token = config.AuthToken.ValueString()
 	}
 
+	if !config.BaseUrl.IsNull() {
+		base_url = config.BaseUrl.ValueString()
+	}
+
+	// Default for API Base URL
+	if base_url == "" {
+		base_url = "https://secure.hosting.de/api/dns/v1/json"
```
Comment (b12f) [inline hostingde/provider.go:117]: It'd be good to have this set as a const at the start of the file

**Review by b12f — CHANGES_REQUESTED** [review]

File: hostingde/provider.go (hunk 1 above)
Comment (nicolindemann) [reply hostingde/provider.go]: Adjusted.

**Review by b12f — APPROVED** [review]

**Comment by b12f** [comment]
Thanks a lot @nicolindemann :)

**Comment by nicolindemann** [comment]
You're welcome @teutat3s and @b12f !

I'm looking forward to using this provider for our use cases too. Any ETA about a new release to the terraform registry?

**Comment by teutat3s** [comment]
> I'm looking forward to using this provider for our use cases too. Any ETA about a new release to the terraform registry?

[New release](https://github.com/pub-solar/terraform-provider-hostingde/releases/tag/v0.0.9) is out! :tada:

**Pull request merged**